core/src/main/java/org/frankframework/validation/SchemaUtils.java

```diff
@@ -223,217 +240,260 @@ private static void addXsdMergeWarnings(final IScopeProvider scopeProvider, fina
 		}
 	}
 
-	public static void xsdToXmlStreamWriter(final IXSD xsd, XMLStreamWriter xmlStreamWriter) throws IOException, ConfigurationException {
-		xsdToXmlStreamWriter(xsd, xmlStreamWriter, true, false, false, false, null, null, null, false);
+	private static @Nonnull Reader createXsdReader(@Nonnull IXSD xsd) throws IOException {
+		Reader reader = xsd.getReader();
+		if (reader == null) {
+			throw new IllegalArgumentException(xsd + " not found");
+		}
+		return reader;
 	}
 
-	private static void xsdToXmlStreamWriter(final IXSD xsd, boolean standalone, boolean stripSchemaLocationFromImport, boolean skipRootStartElement, boolean skipRootEndElement, List<Attribute> rootAttributes, List<Namespace> rootNamespaceAttributes, List<XMLEvent> imports) throws IOException, ConfigurationException {
-		StringWriter dummySchemaContentsWriter = new StringWriter();
-		try {
-			XMLStreamWriter w = XmlUtils.REPAIR_NAMESPACES_OUTPUT_FACTORY.createXMLStreamWriter(dummySchemaContentsWriter);
-			xsdToXmlStreamWriter(xsd, w, standalone, stripSchemaLocationFromImport, skipRootStartElement, skipRootEndElement, rootAttributes, rootNamespaceAttributes, imports, true);
```

Comment:
This method overload has now been renamed to `collectImportsAndSchemaRootAttributes` because that was the purpose.
The loop done in the actual `xsdToXmlStreamWriter` has been inlined into here, and simplified to only execute the logic relevant to this particular use of the method.